01_financial-well-being/income_metrics_county.qmd

```diff
@@ -0,0 +1,735 @@
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Samples",
+       y = NULL)
+```
+
+Read in the PUMA to county crosswalk file. This file is created by the program generate_puma_county_crosswalks.rmd in the geographic-crosswalks folder. 
+```{r}
+puma_county_crosswalk <- read_csv(here::here("geographic-crosswalks", "data", "crosswalk_puma_to_county.csv"))
+```
+
+Crate a version of the crosswalk with just the county-level values of data quality variables for each crosswalk period.
+```{r}
+county_puma <- puma_county_crosswalk %>%
+  group_by(crosswalk_period, statefip, county) %>% 
+  summarise(geographic_allocation_quality = mean(geographic_allocation_quality)) %>% 
+  ungroup()
+```
+
+## Clean Data
+
+Missing data is reported in a variety of ways by IPUMS - listed below:
+
+HHINCOME is missing it is reported as "9999999".
+Vacancy is reported as "0". 
+
+There is a large share of records missing HHINCOME. However, note there are no vacancy results by default of the structure of census data read in.
+
+```{r}
+acs %>% 
+  ggplot(aes(x = hhincome)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Household Income Variable Results") +
+  ylab("Density")
+
+acs %>% 
+  ggplot(aes(x = vacancy)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Vacancy Variable Results") +
+  ylab("Density")
+```
+
+Turn the missing/"not in universe" values for the hhincome variable into NAs.
+
+```{r}
+
+acs_clean <- acs %>%
+  mutate(hhincome = ifelse(hhincome == 9999999, NA_integer_,
+                           hhincome)
+         )
+```
+
+Look at distribution of HHINCOME after adjustment. Some outliers will still exist but bunching around the missing value should not be present.
+
+```{r}
+
+acs_clean %>% 
+  ggplot(aes(x = hhincome)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Household Income Variable Results") +
+  ylab("Density")
+
+```
+
+Select our group quarters, we want to keep only households including additional households under updated definitions.
+
+```{r}
+acs_clean %>% 
+  group_by(gq) %>% 
+  count()
+```
+
+```{r}
+acs_clean <- acs_clean %>% 
+  filter(gq %in% c("Households under 1970 definition", 
+                   "Additional households under 1990 definition",
+                   "Additional households under 2000 definition"))
+
+
+```
+
+Check that group quarters fall only into household categories.
+```{r}
+acs_clean %>% 
+  count(gq) %>% 
+  ggplot(mapping = aes(x = gq, y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("GQ, All Records") +
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "GQ",
+       y = NULL)
+```
+
+### Create race variable
+
+Evaluate the values in the RACE and HISPAN variables.
```

Comment:
From the xml document: "Beginning in 2000, the race question changed substantially to allow respondents to report as many races as they felt necessary to describe themselves. In earlier years, only one race response was coded. Beginning in 2020, the Census Bureau updated the questionnaire text, processing, and coding of the race and Hispanic origin questions, resulting in major changes to the distribution of race and Hispanic origin categories. As a result, users should proceed with caution when comparing RACE and HISPAN in 2019-prior samples with 2020-onward samples. See the comparability tab for more details. " I think that this evaluates race on either side of this cutoff. Not sure where this 'comparability' tab is but might be worth checking out.